### PR TITLE
pkg/nemesis: Add netem chaos supports

### PR DESCRIPTION
**Pull request by mapleFU**
### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/nemesis/netem_chaos.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+		return corrupt{}
+	default:
+		panic("selectNetem received an unexists tag")
+	}
+}
+
+// Generate will randomly generate a chaos without selecting nodes.
+func (g netemChaosGenerator) Generate(nodes []cluster.Node) []*core.NemesisOperation {
+	nChaos := selectNetem(g.name)
+	ops := make([]*core.NemesisOperation, 1)
```
Comment (mahjonp) [inline pkg/nemesis/netem_chaos.go]: replace 1 with len(nodes), Generate requires returning a len(nodes) operation array

File: pkg/nemesis/netem_chaos.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	case "duplicate":
+		return duplicate{}
+	case "corrupt":
+		return corrupt{}
+	default:
+		panic("selectNetem received an unexists tag")
+	}
+}
+
+// Generate will randomly generate a chaos without selecting nodes.
```
Comment (you06) [inline pkg/nemesis/netem_chaos.go]: Does the comment seems strange? It says "randomly generate a chaos", but it's not random, it composes chaos from given name.

File: pkg/nemesis/netem_chaos.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+func (g netemChaosGenerator) Generate(nodes []cluster.Node) []*core.NemesisOperation {
+	nChaos := selectNetem(g.name)
+	ops := make([]*core.NemesisOperation, 1)
+
+	ops[0] = &core.NemesisOperation{
```
Comment (you06) [inline pkg/nemesis/netem_chaos.go]: Is `core.NemesisOperation` strange here? If you call Name function of it, it's "noop".
https://github.com/pingcap/tipocket/blob/dev-netem-chaos/pkg/core/nemesis.go#L112

Comment (mapleFU) [reply pkg/nemesis/netem_chaos.go]: I need to extract nChaos from Operation.

File: pkg/nemesis/netem_chaos.go (hunk 1 above)
Comment (mapleFU) [reply pkg/nemesis/netem_chaos.go]: I'll done it.

File: pkg/nemesis/netem_chaos.go (hunk 2 above)
Comment (mapleFU) [reply pkg/nemesis/netem_chaos.go]: Yes...currently the random comes from runtime, I will change the

File: pkg/nemesis/netem_chaos.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+func (c corrupt) netemType() chaosv1alpha1.NetworkChaosAction {
+	return chaosv1alpha1.CorruptAction
+}
+
+func (c corrupt) template(ns string, pods []string, podMode chaosv1alpha1.PodMode, args ...string) chaosv1alpha1.NetworkChaosSpec {
+	if len(args) != 2 {
+		panic("args number error")
+	}
+	return chaosv1alpha1.NetworkChaosSpec{
+		Action: chaosv1alpha1.DuplicateAction,
```
Comment (mahjonp) [inline pkg/nemesis/netem_chaos.go]:
```suggestion
		Action: chaosv1alpha1.CorruptAction,
```

File: pkg/nemesis/netem_chaos.go (hunk 2 above)
Comment (mapleFU) [reply pkg/nemesis/netem_chaos.go]: Done

File: pkg/nemesis/netem_chaos.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+// Generate will randomly generate a chaos without selecting nodes.
+func (g netemChaosGenerator) Generate(nodes []cluster.Node) []*core.NemesisOperation {
+	nChaos := selectNetem(g.name)
+	ops := make([]*core.NemesisOperation, len(nodes))
+
+	for _, node := range nodes {
+		ops = append(ops, &core.NemesisOperation{
+			Type:        core.NetemChaos,
+			InvokeArgs:  []interface{}{node, nChaos},
```
Comment (mahjonp) [inline pkg/nemesis/netem_chaos.go]: We can save target node in the `Node` field of NemesisOperation here.

**Pull request merged**